TypeCobol.LanguageServer/TypeCobolLanguageServer.cs

```diff
@@ -278,6 +282,49 @@ public void WholeDocumentChanged(object sender, EventArgs eventArgs)
             UpdateTokensLines(compilationDocument);
         }
 
+        /// <summary>
+        /// LSR Cfg Analyzer ID.
+        /// </summary>
+        internal const string lspcfgId = "lsp-cfg";
+
+        /// <summary>
+        /// Method to update CFG/DFA information.
+        /// </summary>
+        /// <param name="fileCompiler">The underlying File Compiler</param>
+        /// <param name="writeToFile">True if CfgDfaParams shall write the dot content to a temporary file, 
+        /// false if the dot content shall be put in the CfgDfaParams.dotContent field.</param>
+        /// <returns>CFG/DFA Data information</returns>
+        public CfgDfaParams UpdateCfgDfaInformation(DocumentContext docContext, bool writeToFile)
+        {
+            CfgDfaParams result = null;
+            docContext.FileCompiler.CompilationResultsForProgram.TryGetAnalyzerResult(lspcfgId, out IList<ControlFlowGraph<Node, object>> cfgs);
+            if (cfgs != null && cfgs.Count > 0)
+            {                
+                //Create a temporary dot file.
+                string tempFile = Path.GetTempFileName();
+                using (TextWriter writer = writeToFile ? File.CreateText(tempFile)  : (TextWriter)new StringWriter())
+                {
+                    CfgDfaParamsBuilder builder = new CfgDfaParamsBuilder(new TextDocumentIdentifier(docContext.TextDocument.uri), writeToFile ? tempFile : null);
+                    CfgDotFileForNodeGenerator<object> gen = new CfgDotFileForNodeGenerator<object>(cfgs[0]);
+                    gen.FullInstruction = true;
+                    gen.BlockEmittedEvent += (block, subgraph) => builder.AddBlock<object>(block, subgraph);
+                    gen.Report(writer);
+                    result = builder.GetParams();
+                    if (!writeToFile)
+                    {
+                        writer.Flush();
```

Comment:
`writer.Flush()` is not required. `StringWriter` has empty implementation for it.

Reply:
As writer is declared as a `TextWriter` I'd rather keep `writer.Flush();` just in case the implementation change.

Reply:
In fact, my justification was wrong. The next line `result.dotContent = writer.ToString();` assumes that the writer is a `StringWriter` because only `StringWriter` redefines its `ToString()` method to return all characters accumulated so far. So the `writer.Flush();` call is unnecessary because `ToString` will flush all data.

Reply:
It ssems to  that this Flush was necessay tha't s why i Put it.

Reply:
I think you are right, it seems that internal representation of StringWriter uses a StringBuilder.
https://referencesource.microsoft.com/#mscorlib/system/io/stringwriter.cs